src/Cli/dotnet/Commands/Restore/RestoreCommand.cs

```diff
@@ -51,4 +63,26 @@ public static int Run(ParseResult parseResult)
 
         return FromParseResult(parseResult).Execute();
     }
+
+    public abstract int Execute();
+}
+
+public sealed class ForwardingRestoreCommand : RestoreCommand
+{
+    public ForwardingRestoreCommand(IEnumerable<string> msbuildArgs, string msbuildPath = null)
```

Comment:
I suspect I'm missing something, but I don't fully understand why we need to wrap the MSBuildForwardingApp in a command. I assume the type hierarchy requires it, but it seems like it should be possible to just have an 'executable command' or something that implements `Execute` and then we wouldn't need as many type wrappers?

Reply:
Yes, with some refactoring we can do that, let me try it.